src/base/bittorrent/session.cpp

```diff
@@ -4513,6 +4514,43 @@ void Session::startUpTorrents()
         if (needStore)
              m_resumeDataStorage->store(torrentID, resumeData);
 
+        const QString category = resumeData.category;
+        bool isCategoryRecovered = recoveredCategories.contains(category);
+        if (!category.isEmpty() && (isCategoryRecovered || !m_categories.contains(category)))
+        {
+            if (!isCategoryRecovered)
+            {
+                if (addCategory(category))
+                {
+                    recoveredCategories.insert(category);
+                    isCategoryRecovered = true;
+                    LogMsg(tr("Inconsistent data is detected. Category '%2' is assigned to some torrent(s) but doesn't exist in configuration file."
+                              " Its settings will be reset to default.").arg(torrentID.toString(), category), Log::WARNING);
+                }
+                else
+                {
+                    resumeData.category.clear();
+                    LogMsg(tr("Inconsistent data is detected. Torrent '%1' has assigned to category '%2' which is invalid.")
```

Comment:
>Torrent '%1' has assigned to category '%2'

I would say "Category '%1' is assigned to torrent '%2'" if you don't have a problem with having Torrent as 2nd argument.
Otherwise i would just remove "to" which is definitely wrong in this sentence. And i'm not entirely sure "has assigned category" is correct either (maybe it is).

Reply:
> I would say "Category '%1' is assigned to torrent '%2'" if you don't have a problem with having Torrent as 2nd argument.

Sorry, how might the whole sentence look like?

Reply:
> > I would say "Category '%1' is assigned to torrent '%2'" if you don't have a problem with having Torrent as 2nd argument.
> 
> Sorry, how might the whole sentence look like?

Sorry scratch that, i misunderstood. The category is invalid not the assignment, so:


```suggestion
                    LogMsg(tr("Inconsistent data is detected. Invalid category '%1' is assigned to torrent '%2'.")
```

Reply:
All done.